File: src/main/java/com/baulsupp/oksocial/apidocs/ApiDocPresenter.java
package com.baulsupp.oksocial.apidocs;

import ee.schimke.oksocial.output.OutputHandler;
import java.io.IOException;
import okhttp3.OkHttpClient;

public interface ApiDocPresenter {
  ApiDocPresenter NONE =
      (url, outputHandler, client) -> outputHandler.info("No documentation for: " + url);

  void explainApi(String url, OutputHandler outputHandler, OkHttpClient client) throws IOException;
}

File: src/main/java/com/baulsupp/oksocial/apidocs/ApiDocPresenter.kt
package com.baulsupp.oksocial.apidocs

import com.baulsupp.oksocial.output.OutputHandler
import java.io.IOException
import okhttp3.OkHttpClient

interface ApiDocPresenter {

    @Throws(IOException::class)
    fun explainApi(url: String, outputHandler: OutputHandler<*>, client: OkHttpClient)

    companion object {
        val NONE = { url, outputHandler, client -> outputHandler.info("No documentation for: " + url) }
    }
}
